cmake/modules/FindUSD.cmake

```diff
@@ -66,6 +66,21 @@ elseif(USD_INCLUDE_DIR AND EXISTS "${USD_INCLUDE_DIR}/pxr/pxr.h")
     math(EXPR PXR_VERSION "${USD_MAJOR_VERSION} * 10000 + ${USD_MINOR_VERSION} * 100 + ${USD_PATCH_VERSION}")
 endif()
 
+# Set special Autodesk USD version. We use this to communicate whether or not
+# there are extra patches on-top of the normal USD build.
+set(ADSK_USD_VERSION ${USD_VERSION})
+if (USD_VERSION VERSION_LESS_EQUAL "0.23.02")
+    # In the security fix there was a new inline method "max_size" added so we
+    # can use that to determine if this USD build has the security fix.
+    if (USD_INCLUDE_DIR AND EXISTS "${USD_INCLUDE_DIR}/pxr/base/vt/array.h")
+        file(STRINGS ${USD_INCLUDE_DIR}/pxr/base/vt/array.h USD_HAS_SECURITY_PATCH REGEX "constexpr size_t max_size\(\)")
+        if (USD_HAS_SECURITY_PATCH)
+            set(ADSK_USD_VERSION "${ADSK_USD_VERSION}-ad1")
+            message(STATUS "USD v${USD_VERSION} has security patches.")
+        endif()
+    endif()
+endif()
```

Comment:
Added new cmake variable "ADSK_USD_VERSION" which starts out equal to normal "USD_VERSION". Detect in an automatic way if the USD build (any version <= 0.23.02) has the extra security patches. Do this by looking at the vt/array.h header file for a new method which was added in one of the patches. If so then add "-ad1" to new variable ADSK_USD_VERSION.